presto-main/src/main/java/com/facebook/presto/StatisticalDigest.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto;
+
+import io.airlift.slice.Slice;
+
+public interface StatisticalDigest<T>
+{
+    default void add(double value, long weight)
+    {
+        throw new UnsupportedOperationException("Cannot add a double to a q-digest");
```

Comment:
I would remove the message entirely, as the exception type indicates it already.